include/boost/fusion/container/set/set.hpp

```diff
@@ -97,39 +97,20 @@ namespace boost { namespace fusion
 
         template <typename Sequence>
         BOOST_FUSION_GPU_ENABLED
-        set(Sequence const& rhs)
-            : data(rhs) {}
-
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
-        template <typename Sequence>
-        BOOST_FUSION_GPU_ENABLED
         set(Sequence&& rhs)
             : data(std::forward<Sequence>(rhs)) {}
-#endif
 
         BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(typename detail::call_param<T>::type ...args)
             : data(args...) {}
 
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
         template <typename ...U>
         BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(U&& ...args)
```

Comment:
`template<typename Sequence> set(Sequence&&)` is selected whenever `fusion::set` construction is given one argument, and is implicitly constructible from one argument without restrictions. The C++03 version uses `disable_if<is_convertible<Sequence, T0>>`, making it implicitly constructible only from another sequence. I'm guessing this is desired.

T0 doesn't exist. I _think_ a better expression is `enable_if<is_sequence<Sequence> && sizeof...(T) == Sequence::size>`. Except, make it not error when `Sequence` isn't a fusion sequence. Its tempting to leave out the size comparison expression, but its omission would allow implicit conversions like `fusion::vector<int, float>` to `fusion::set<fusion::vector<int, float>>`. I think that would be a bug (a two element sequence implicitly converting to a 1 element sequence). However, explicit construction should be allowed in that scenario, which is still provided by the `U&&...` constructor.

The `U&&` constructor should not need explicit disabling anymore (I was wrong earlier!), since it is lower precedence to the single argument constructor.